services/provider/server/server_test.go

```diff
@@ -0,0 +1,343 @@
+	},
+	{
+		Name: "rook-ceph-client-cephuser-rbd-provisioner",
+		Kind: "Secret",
+		Data: map[string]string{
+			"userID":  "cephuser-rbd-provisioner",
+			"userKey": "AQADw/hhqBOcORAAJY3fKIvte++L/zYhASjYPQ==",
+		},
+	},
+	{
+		Name: "rook-ceph-client-cephuser-rbd-node",
+		Kind: "Secret",
+		Data: map[string]string{
+			"userID":  "cephuser-rbd-node",
+			"userKey": "AQADw/hhqBOcORAAJY3fKIvte++L/zYhASjYPQ==",
+		},
+	},
+	{
+		Name: "rook-ceph-client-cephuser-cephfs-provisioner",
+		Kind: "Secret",
+		Data: map[string]string{
+			"adminID":  "cephuser-cephfs-provisioner",
+			"adminKey": "AQADw/hhqBOcORAAJY3fKIvte++L/zYhASjYPQ==",
+		},
+	},
+	{
+		Name: "rook-ceph-client-cephuser-cephfs-node",
+		Kind: "Secret",
+		Data: map[string]string{
+			"adminID":  "cephuser-cephfs-node",
+			"adminKey": "AQADw/hhqBOcORAAJY3fKIvte++L/zYhASjYPQ==",
+		},
+	},
+	{
+		Name: "rook-ceph-client-cephuser-health-checker",
+		Kind: "Secret",
+		Data: map[string]string{
+			"userID":  "cephuser-health-checker",
+			"userKey": "AQADw/hhqBOcORAAJY3fKIvte++L/zYhASjYPQ==",
+		},
+	},
+}
+
+var (
+	consumerResource = &ocsv1alpha1.StorageConsumer{
+		ObjectMeta: metav1.ObjectMeta{Name: "consumer", UID: "uid"},
+		Status: ocsv1alpha1.StorageConsumerStatus{
+			CephResources: []ocsv1alpha1.CephResourcesSpec{
+				{
+					Name: "cephblockpool",
+					Kind: "CephBlockPool",
+					CephClients: map[string]string{
+						"node":        "cephuser-rbd-node",
+						"provisioner": "cephuser-rbd-provisioner",
+					},
+				},
+				{
+					Name: "cephFilesystemSubVolumeGroup",
+					Kind: "CephFilesystemSubVolumeGroup",
+				},
+				{
+					Name: "cephuser-rbd-provisioner",
+					Kind: "CephClient",
+				},
+				{
+					Name: "cephuser-rbd-node",
+					Kind: "CephClient",
+				},
+				{
+					Name: "cephuser-cephfs-provisioner",
+					Kind: "CephClient",
+				},
+				{
+					Name: "cephuser-cephfs-node",
+					Kind: "CephClient",
+				},
+				{
+					Name: "cephuser-health-checker",
+					Kind: "CephClient",
+				},
+			},
+			State: ocsv1alpha1.StorageConsumerStateReady,
+		},
+	}
+
+	consumerResource1 = &ocsv1alpha1.StorageConsumer{
+		ObjectMeta: metav1.ObjectMeta{Name: "consumer1", UID: "uid1"},
+		Status: ocsv1alpha1.StorageConsumerStatus{
+			State: ocsv1alpha1.StorageConsumerStateFailed,
+		},
+	}
+
+	consumerResource2 = &ocsv1alpha1.StorageConsumer{
+		ObjectMeta: metav1.ObjectMeta{Name: "consumer2", UID: "uid2"},
+		Status: ocsv1alpha1.StorageConsumerStatus{
+			State: ocsv1alpha1.StorageConsumerStateConfiguring,
+		},
+	}
+	consumerResource3 = &ocsv1alpha1.StorageConsumer{
+		ObjectMeta: metav1.ObjectMeta{Name: "consumer3", UID: "uid3"},
+		Status: ocsv1alpha1.StorageConsumerStatus{
+			State: ocsv1alpha1.StorageConsumerStateDeleting,
+		},
+	}
+	consumerResource4 = &ocsv1alpha1.StorageConsumer{
+		ObjectMeta: metav1.ObjectMeta{Name: "consumer4", UID: "uid4"},
+		Status:     ocsv1alpha1.StorageConsumerStatus{},
+	}
+)
+
+func TestGetJSONConnectionDetails(t *testing.T) {
```

Comment:
I think we need to change the name of this test, the current name refer to something that does not exists anymore